app/src/main/java/org/mozilla/focus/broadcastreceiver/DownloadBroadcastReceiver.java

```diff
@@ -0,0 +1,99 @@
+/* -*- Mode: Java; c-basic-offset: 4; tab-width: 4; indent-tabs-mode: nil; -*-
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.broadcastreceiver;
+
+import android.app.DownloadManager;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.content.pm.ResolveInfo;
+import android.database.Cursor;
+import android.net.Uri;
+import android.support.design.widget.Snackbar;
+import android.support.v4.content.ContextCompat;
+import android.support.v4.content.FileProvider;
+import android.view.View;
+import android.webkit.MimeTypeMap;
+import android.webkit.URLUtil;
+
+import org.mozilla.focus.BuildConfig;
+import org.mozilla.focus.R;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.List;
+
+public class DownloadBroadcastReceiver extends BroadcastReceiver {
+    private final View browserContainer;
+    private final static HashMap<Long, DownloadManager> referenceIds = new HashMap<>();
+
+    public DownloadBroadcastReceiver(View view) {
+        this.browserContainer = view;
+    }
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        final long mDownloadReference = intent.getLongExtra(DownloadManager.EXTRA_DOWNLOAD_ID, -1);
+        displaySnackBar(context, mDownloadReference);
+    }
+
+    public void displaySnackBar(final Context context, long downloadReference) {
```

Comment:
nit: it's confusing what the diff between `referenceIds` and `downloadReference` is. Perhaps change them (respectively to): `queuedDownloadReferences` and `completedDownloadReference`.